### PR TITLE
(SDI-1977) Fixes NaN errors

### PR DESCRIPTION
**Pull request by marea-cobb**
Fixes #

Summary of changes:
- Removed NaN errors occurring during marshalling.

How to verify it:
- make test-small

Testing done:
- make test-small

A picture of a snapping turtle (not required but encouraged):
-

### CODE REVIEW

File: statistics/statistics.go [hunk 1]
```diff
@@ -63,10 +63,12 @@ func Meta() *plugin.PluginMeta {
...
 func New() *Plugin {
 	buffer := make(map[string][]interface{})
+	curSize := make(map[string]int)
+	bufIdx := make(map[string]int)
 	p := &Plugin{buffer: buffer,
 		bufferMaxSize: 100,
-		bufferCurSize: 0,
-		bufferIndex:   0}
+		bufferCurSize: curSize,
+		bufferIndex:   bufIdx}
```
Comment (candysmurf) [inline statistics/statistics.go]: index or map? if it's a map, rename it

File: statistics/statistics.go [hunk 2]
```diff
@@ -241,27 +247,26 @@ func concatNameSpace(namespace []string) string {
...
+		p.bufferCurSize[ns]++
 	}
 
-	if p.bufferIndex == p.bufferMaxSize-1 {
-		p.bufferIndex = 0
+	if p.bufferIndex[ns] == p.bufferMaxSize-1 {
+		p.bufferIndex[ns] = 0
 	} else {
-		p.bufferIndex++
+		p.bufferIndex[ns]++
```
Comment (candysmurf) [inline statistics/statistics.go]: I never understand why you need another index here.

File: statistics/statistics.go (hunk 1 above)
Comment (marea-cobb) [inline statistics/statistics.go]: This is a map storing buffer indices... Would a better name be bufferIndexMap?

File: statistics/statistics.go (hunk 2 above)
Comment (marea-cobb) [inline statistics/statistics.go]: I don't understand your question. Are you asking why it needs to be a map or why we increment the index here?

Comment (candysmurf) [inline statistics/statistics.go]: It's OK. I just don't think the algorithm should be like this. No worries.

**Comment by candysmurf** [comment]
@marea-cobb, do you like to get your build issue fix before we get this merged?

**Comment by marea-cobb** [comment]
It looks like statistics.go failing the gofmt test. I can't fix this however because I no longer have an environment setup to run the file through. I made the fixes straight through the GitHub GUI. It will be a lot of trial and error trying to find the exact spacing issue.

**Comment by candysmurf** [comment]
@marea-cobb, no issue. I'll try to fix it.

**Comment by candysmurf** [comment]
As we're refactoring the code base and remove the third party library used. This PR is obsolete.

**Pull request closed without merging**